OFEP-ofo-flagd-client-support.md

```diff
@@ -0,0 +1,36 @@
+## flagd client support driven by OpenFeature Operator
+
+## State: DRAFTING
+
+Client side applications (e.g. web/mobile) could benefit from access to feature flags. This OFEP outlines the feasibility and benefits of extending [OpenFeature Operator (OFO)](https://github.com/open-feature/open-feature-operator) to manage the deployment of flagd with access from external traffic (e.g. client side applications).
+
+## Background
+
+OFO already manages the deployment of flagd in server side contexts, handling the networking necessary for server side applications to communicate with flagd. This greatly simplifies the steps necessary for administrators to introduce flagd into their cluster, reducing the barrier of entry. Obtaining these benefits for the deployment of externally accessible flagd is the critical motivator for extending OFO.
+
+## Assumptions
+The (simplified) deployment pattern is as follows
+
+- Deployment of a client side application exposed by a Service
+- [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) exposing the app's Service via a [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/) and listener
+
+The networking infrastructure for ingesting external traffic is already in place for the existing client side application.
+
+## Proposal
+
+Introduce a custom resource definition (CRD) with configuration permitting OFO to create a deployment of flagd and inject the necessary networking configuration to route incoming traffic from the existing Gateway to flagd (via a new HTTPRoute/GRPCRoute).
```

Comment:
Can you walk us through a bit more detail on the flow here?

Would the OFO need CRUD for routes on all namespaces in this scenario?

If I create a new deployment, with a flagD sidecar and have a route for my application to the wider world, does this proposal create a second route for the feature flag API?

Reply:
> Would the OFO need CRUD for routes on all namespaces in this scenario?

It'd need CRUD on the namespace of the existing Gateway. How we restrict OFO to just that namespace is something to think about, I will reflect on this.

> If I create a new deployment, with a flagD sidecar and have a route for my application to the wider world, does this proposal create a second route for the feature flag API?

Yes, in that scenario the flagd sidecar is closed to access from the wider world. A new flagd deployment is created with an exposed (to the wider world) route to that flagd instance.